src/frontend/src/components/MapComponent/OpenLayersComponent/LayerSwitcher/index.js

```diff
@@ -7,8 +7,8 @@ import LayerTile from 'ol/layer/Tile';
 import SourceOSM from 'ol/source/OSM';
 import LayerSwitcher from 'ol-layerswitcher';
 import React, { useEffect } from 'react';
-
 import { XYZ } from 'ol/source';
+import { useLocation } from 'react-router-dom';

```

Comment:
this is a seperate dependent added onto the isolated mapcomponent @NSUWAL123 please replace this logic and make it pass through props instead of using it in this component.